theano/tensor/nnet/abstract_conv.py

```diff
@@ -2212,18 +2213,23 @@ def perform(self, node, inp, out_):
 
         # Rearranging the weights into the correct shape for unshared
         if self.unshared is True:
-            for i in range(0, kern.shape[2]):
-                for j in range(0, kern.shape[3]):
-                    for m in range(0, kern.shape[4]):
-                        for n in range(0, kern.shape[5]):
-                            if i + m > kern.shape[2] or j + n > kern.shape[3]:
-                                kern[:, :, i, j, m, m] = 0
+            new_kshp = (kern.shape[0], kern.shape[1],
+                        kern.shape[4] + topgrad.shape[2] - 1,
+                        kern.shape[5] + topgrad.shape[3] - 1,
+                        kern.shape[4], kern.shape[5])
+            new_kern = np.zeros((new_kshp), dtype=topgrad.dtype)
+            for i in range(0, new_kshp[2]):
+                for j in range(0, new_kshp[3]):
+                    for m in range(0, new_kshp[4]):
+                        for n in range(0, new_kshp[5]):
+                            if i + m >= kern.shape[2] or j + n >= kern.shape[3]:
+                                new_kern[:, :, i, j, m, n] = 0
                             else:
-                                kern[:, :, i, j, m, m] = kern[:, :, i + m, j + n, m, n]
+                                new_kern[:, :, i, j, m, n] = kern[:, :, i + m, j + n, m, n]
```

Comment:
Does this work? It's somewhat complex, so I might have made a mistake in my manual derivations, but I think this is missing the first part of the kernel and shifting everything else too far to the left.

In a 1D case, with a 5-pixel input and 3-pixel kernel, I think you want the kernel to be changed like this:
```
kern = [[a, b, c], [d, e, f], [g, h, i]]
new_kern = [[_, _, a], [_, b, d], [c, e, g], [f, h, _], [i, _, _]]
```
but this implementation would change it to this, I think:
```
new_kern = [[a, b, c], [d, e, _], [g, _, _], [_, _, _], [_, _, _]]
```